src/System.Xml.XmlDocument/tests/Performance/Perf.XmlNodeList.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Xml;
+using Microsoft.Xunit.Performance;
+
+namespace XmlDocumentTests.XmlNodeListTests
+{
+    public class Perf_XmlNodeList
+    {
+        [Benchmark]
+        public void GetEnumerator()
+        {
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                // Setup
+                var doc = new XmlDocument();
+                doc.LoadXml("<a><sub1/><sub2/></a>");
+                XmlNodeList list = doc.DocumentElement.ChildNodes;
+
+                // Actual perf testing
+                using (iteration.StartMeasurement())
+                {
+                    list.GetEnumerator(); list.GetEnumerator(); list.GetEnumerator();
+                    list.GetEnumerator(); list.GetEnumerator(); list.GetEnumerator();
+                    list.GetEnumerator(); list.GetEnumerator(); list.GetEnumerator();
```

Comment:
For completeness sake I added tests for all items from the usage data that were highly utilized. I tried to filter out the obviously unnecessary ones, but as you noticed I missed some :) I took a very broad approach to my first-pass to get some tests out into the open for other people to start playing around with, but it's likely that ~20% of these tests will be removed for one reason or another (e.g. the field tests). 

The GetEnumerator tests are instances of this as well. GetEnumerator <i>is</i> what I intended to test, but the question of whether or not I <i>should</i> test it slipped by me. Now that these tests are in the open, I need to go back through them and for each test ask: <i>is this really what I care about performance for?</i> In the case of GetEnumerator I think your reasoning is correct that if GetEnumerator is highly utilized then we should perf test it, but GetEnumerator is frequently enough accompanied by iteration that we can instead perf test that.